video/out/gpu/video.c

```diff
@@ -1727,6 +1727,12 @@ static void reinit_scaler(struct gl_video *p, struct scaler *scaler,
 
     uninit_scaler(p, scaler);
 
+    if (conf && scaler->index == SCALER_DSCALE && (!conf->kernel.name || 
+        !strcmp(conf->kernel.name, "")))
+    {
+        conf = &p->opts.scaler[SCALER_SCALE];
+    }
```

Comment:
also shouldn't this go to the very top of the function?

Reply:
`scaler_conf_eq` will return false in the case of an empty name, so I don't think it needs to be at the very top.